src/main/java/duke/DukeList.java

```diff
@@ -0,0 +1,258 @@
+package duke;
+
+
+import duke.exceptions.DukeException;
+import duke.exceptions.DukeInvalidCommandException;
+import duke.exceptions.DukeInvalidDescriptionException;
+import duke.exceptions.DukeNoDateException;
+import duke.exceptions.DukeNoDescriptionException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.TaskType;
+import duke.tasks.Todo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * DukeList class for Duke.
+ * Stores tasks.
+ */
+public class DukeList {
+
+    private static final int CAPACITY = 100;
+    private final List<Task> taskList;
+    private final Storage store = new Storage("data", "duke.txt");
+
+
+    /**
+     * Constructor for DukeList.
+     */
+    public DukeList() {
+        this.taskList = new ArrayList<>(CAPACITY);
+    }
+
+
+    /**
+     * Constructor for DukeList.
+     * Overrides default capacity
+     *
+     * @param capacity list capacity.
+     */
+    public DukeList(int capacity) {
+        this.taskList = new ArrayList<>(capacity);
+    }
+
+
+    /**
+     * Helper function for adding item.
+     *
+     * @param descriptionString String to be added.
+     * @return Task added.
+     * @throws DukeInvalidDescriptionException invalid description.
+     * @throws DukeInvalidCommandException     invalid command.
+     */
+    private Task addHelper(String descriptionString) throws DukeInvalidDescriptionException, DukeInvalidCommandException {
+        Task newTask;
+
+        String[] strArr = Parser.parseLineToArray(descriptionString);
+        TaskType taskType = Parser.getTaskKeyword(descriptionString);
+
+        String formattedDescriptionString;
+
+        switch (taskType) {
+        case TODO:
+            try {
+                formattedDescriptionString = Parser.getItemSubstring(strArr);
+                newTask = new Todo(formattedDescriptionString);
+                break;
+            } catch (DukeNoDescriptionException e) {
```

Comment:
Maybe the message here ("The description cannot be empty") can be encapsulated in DukeNoDescriptionException instead, so you don't have to throw a new Exception.